pandas/core/reshape/merge.py

```diff
@@ -1970,7 +1985,7 @@ def _any(x) -> bool:
     return x is not None and com.any_not_none(*x)
 
 
-def validate_operand(obj):
+def _validate_operand(obj: FrameOrSeries) -> ABCDataFrame:
```

Comment:
ABCDataFrame resolves to Any, use DataFrame and import using if TYPE_CHECKING if necessary.

Reply:
Will do.  Is that resolution something we'll be able to fix eventually or just how it is?

Reply:
I think a solution could be to use cast in `core.dtypes.generic`

i.e `ABCDataFrame = cast(DataFrame, create_pandas_abc_type("ABCDataFrame", "_typ", ("dataframe",)))`

I'll look into it.

If we can live with the noise in `core.types.generic` it will reduce the number of `if TYPE_CHECKING` and `# noqa` needed in general.

Reply:
that doesn't seem to work since the cast is on the type not an instance and just results in lots of `Variable "pandas.core.dtypes.generic.ABC..." is not valid as a type` errors.